ComplementaryScripts/modelCuration/AddCofactorIon.m

```diff
@@ -0,0 +1,216 @@
+[m, n]      = size(CONValldata);
+for i = 1:m
+    aa = CONValldata(i,1);
+    aa = char(aa);
+    for j=1:length(matrix.rxnIDs)
+        bb = matrix.metcompartments(j,1);
+        bb = char(bb);
+        if strcmp(bb,aa)
+            matrix.Newcomps(j,1) = CONValldata(i,2);
+        end
+    end
+end
+for i=1:length(matrix.rxnIDs)
+    matrix.metnames(i) = strcat(matrix.metIDs(i),lbracket,matrix.metcompartments(i),rbrackets);
+    matrix.Newcomps(i) = strcat(llbracket,matrix.Newcomps(i),rbrackets);
+end
+
+% Map mets to model.metnames, get s_index for new mets:
+cd otherChanges
+for j = 1:length(matrix.metnames)
+    [~,metindex] = ismember(matrix.metnames(j),model.metNames);
+    if metindex ~= 0
+        matrix.mets(j) = model.mets(metindex);
+    elseif metindex == 0
+        newID = getNewIndex(model.mets);
+        matrix.mets(j) = strcat('s_',newID,matrix.Newcomps(j));
+        model = addMetabolite(model,char(matrix.mets(j)), ...
+            'metName',matrix.metnames(j));
+    end
+end
+
+% Add metabolite data:
+fid = fopen('../../ComplementaryData/modelCuration/Biomass_newRxnMet.tsv');
+newmet_annot         = textscan(fid,'%s %s %s %s %s %s %s','Delimiter','\t','HeaderLines',1);
+newmet.metNames      = newmet_annot{1};
+newmet.metFormulas   = newmet_annot{2};
+newmet.metCharges    = cellfun(@str2num, newmet_annot{3});
+newmet.metKEGGID     = newmet_annot{5};
+newmet.metChEBIID    = newmet_annot{6};
+newmet.metMetaNetXID = newmet_annot{7};
+
+fclose(fid);
+for i = 1:length(newmet.metNames)
+    [~,metID] = ismember(newmet.metNames(i),model.metNames);
+    if metID ~= 0
+        model.metFormulas{metID}   = newmet.metFormulas{i};
+        model.metCharges(metID)    = newmet.metCharges(i);
+        model.metKEGGID{metID}     = newmet.metKEGGID{i};
+        model.metChEBIID{metID}    = newmet.metChEBIID{i};
+        model.metMetaNetXID{metID} = newmet.metMetaNetXID{i};
+        model.metNotes{metID}      = 'NOTES: added for BiomassUpdate';
+    end
+end
+
+% Add new reactions according to rev ID: Met Coef needs to be a column, not
+% a row. Coef should be a double, which was converted at the import section
+EnergyResults     = {};
+MassChargeresults = {};
+RedoxResults      = {};
+if ~isfield(model,'rxnMetaNetXID')
+    model.rxnMetaNetXID = cell(size(model.rxns));
+end
+for i = 1:length(newrxn.ID)
+    newID = getNewIndex(model.rxns);
+    j     = find(strcmp(matrix.rxnIDs,newrxn.ID{i}));
+    Met   = matrix.mets(j);
+    Coef  = transpose(matrix.metcoef(j));
+    [model,rxnIndex] = addReaction(model, ['r_' newID],...
+        'reactionName', newrxn.ID{i},...
+        'metaboliteList',Met,...
+        'stoichCoeffList',Coef,...
+        'reversible',newrxn.Rev(i,1),...
+        'geneRule',newrxn.GPR{i},...
+        'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
```

Comment:
Lines 131 and 132 throw error: `'CheckEnergyProduction' is not found in the current folder or on the MATLAB path, but exists in: C:\Users\bensan\Documents\repos\yeast-GEM\ComplementaryScripts\modelCuration`
So before and after you call these functions you should `cd` to the proper locations

Reply:
okay.

Reply:
It appeared that this function should be named as `addBiomassUpdate`, according to the description and script content. The current name `AddCofactorIon` seems misleading in my view.